Update feature gates (English) via automation

singh1203 opened the pull request:
<!--
 Hello!

 PLEASE title the FIRST commit appropriately, so that if you squash all
 your commits into one, the combined commit message makes sense.
 For overall help on editing and submitting pull requests, visit:
  https://kubernetes.io/docs/contribute/suggesting-improvements/

 Use the default base branch, “main”, if you're documenting existing
 features in the English localization.

 If you're working on a different localization (not English), see
 https://kubernetes.io/docs/contribute/new-content/overview/#choose-which-git-branch-to-use
 for advice.

 If you're documenting a feature that will be part of a future release, see
 https://kubernetes.io/docs/contribute/new-content/new-features/ for advice.
-->
### Description
Syncs all feature gate `.md` files with Kubernetes

Generated using: `./scripts/releng/update-feature-gates.sh ../featuregates.json`

**Depends on (merge AFTER):**
- kubernetes/kubernetes#135510
- kubernetes/website#53516
<!--
 Remember to ADD A DESCRIPTION and delete this note before submitting
 your pull request. The description should explain what will change,
 and why.
-->

### Issue

<!--
 If this pull request resolves an open issue, please link the issue in the PR
 description so it will automatically close when the PR is merged.

 See the GitHub documentation for more details and other options:

 https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword
-->

Closes:

singh1203 commented:
Are checks failing due to my changes?
I don't think so https://app.netlify.com/projects/kubernetes-io-main-staging/deploys/695228c6671767000820d9c3#L17

lmktfy reviewed (commented):
Thank you for the PR.

To make this build, try a couple of changes:

* omit all the new feature gates where there is a `TODO` in the description
  (save those for a future pull request)
* don't have a `stable` stage starting at version 1.0 for feature gates that were introduced as `deprecated`; see `AllowDNSOnlyNodeCSR` for an example of this

---

If you [preview the changes locally](https://kubernetes.io/docs/contribute/new-content/open-a-pr/#preview-locally) you can see if there are any build errors.
If there are, it's OK to ask for advice about addressing them.

singh1203 committed:
Update feature gate markdown files based on versioned_feature_list.yaml

Signed-off-by: Saurabh Kumar Singh <singh1203.ss@gmail.com>

singh1203 commented:
> Thank you for the PR.
> 
> To make this build, try a couple of changes:
> 
> * omit all the new feature gates where there is a `TODO` in the description
>   (save those for a future pull request)
> * don't have a `stable` stage starting at version 1.0 for feature gates that were introduced as `deprecated`; see `AllowDNSOnlyNodeCSR` for an example of this
> 
> If you [preview the changes locally](https://kubernetes.io/docs/contribute/new-content/open-a-pr/#preview-locally) you can see if there are any build errors. If there are, it's OK to ask for advice about addressing them.

Updated this PR to include only feature gates tracked in [versioned_feature_list.yaml](https://github.com/kubernetes/kubernetes/blob/master/test/compatibility_lifecycle/reference/versioned_feature_list.yaml) from [k/k](https://github.com/kubernetes/kubernetes)

Changes:
- Removed placeholder feature gates with FIXME/TODO descriptions
- Kept only gates with existing documentation that received front matter updates
- Verified locally with make container-serve - build succeeds ✅

New feature gates requiring descriptions will be handled in a separate PR with proper context.
@lmktfy thank you for guidance and feedback :bow:

lmktfy commented:
/retitle Update feature gates (English) via automation

lmktfy commented on content/en/docs/reference/command-line-tools-reference/feature-gates/DynamicResourceAllocation.md:
```diff
@@ -8,22 +8,20 @@ _build:
     toVersion: "1.33"
   - stage: stable
     defaultValue: true
-    locked: false
```
I would keep this, but for that we ought to first update the generator code.

lmktfy reviewed (commented):
/lgtm

Even though this PR introduces one new error, the overall changes represent a big improvement.

divya-mohan0209 commented:
@singh1203 : Please advise if you'd be able to address comments by @sftim ?

singh1203 commented:
> @singh1203 : Please advise if you'd be able to address comments by @sftim ?

Yes, I'm working on it and need some clarification, as I mentioned in the reply to @lmktfy's comment.

lmktfy commented on content/en/docs/reference/command-line-tools-reference/feature-gates/MutatingAdmissionPolicy.md:
```diff
@@ -6,9 +6,9 @@ _build:
+  - stage: alpha
     defaultValue: false
-    fromVersion: "1.30"
+    fromVersion: "1.32"
```
If the upstream data are wrong, let's fix it in k/k so that the generated update comes out correct.

lmktfy replied:
This feature gate was available to set in Kubernetes 1.30 and Kubernetes 1.31 and I would be very wary to change what we say retrospectively. The docs were already correct.

singh1203 replied:
I investigated `MutatingAdmissionPolicy` in k/k, and the `v1.30/v1.31` history [here](https://github.com/kubernetes/kubernetes/commit/4b13362dda020f80cf961dbeefae750f63f803f9) was deliberately adjusted so the recorded lifecycle starts at `v1.32`. Since that change is intentional upstream, do you still want to **_fix it in k/k so that the generated update comes out correct?_**

singh1203 committed:
Merge branch 'main' into gen_reference_feature_gates_markdown

singh1203 committed:
chore: update feature gates markdown with utility

Signed-off-by: Saurabh Kumar Singh <singh1203.ss@gmail.com>

singh1203 committed:
Merge branch 'main' into gen_reference_feature_gates_markdown

singh1203 commented:
Closing this on behalf of https://github.com/kubernetes/website/pull/55761, as front matter was missing after making changes in the tool. 
Thank you!
